src/reactviews/common/keyboardUtils.ts

```diff
@@ -0,0 +1,284 @@
+        if (modifierHandler) {
+            modifierHandler(matcher, displayTokens);
+            continue;
+        }
+
+        if (keyAssigned) {
+            // Unsupported chord; only single primary key is handled.
+            return undefined;
+        }
+
+        const keyInfo = resolveKeyToken(token);
+        if (!keyInfo) {
+            return undefined;
+        }
+
+        matcher.key = keyInfo.key;
+        if (keyInfo.code) {
+            matcher.code = keyInfo.code;
+        }
+        displayTokens.push(keyInfo.display);
+        keyAssigned = true;
+    }
+
+    if (!keyAssigned) {
+        return undefined;
+    }
+
+    return { keyCombination: matcher, label: displayTokens.join("+") };
+}
+
+/**
+ * Gets the shortcut information from the raw keyboard shortcut string.
+ * @param raw Raw keyboard shortcut string from configuration
+ * @returns Parsed webview shortcut
+ */
+export function getShortcutInfo(raw: string | undefined): WebviewKeyBinding {
+    const primaryTokens = normalize(raw);
+    const primary = buildShortcut(primaryTokens);
+    if (primary) {
+        return primary;
+    }
+
+    return { keyCombination: {}, label: "" };
+}
+
+function getDefaultConfig(): WebviewKeyBindingConfiguration {
+    return {
+        [WebviewAction.ResultGridSelectAll]: "ctrlcmd+a",
+        [WebviewAction.ResultGridCopySelection]: "ctrlcmd+c",
+    } as WebviewKeyBindingConfiguration;
+}
+
+/**
+ * Parses the webview keyboard shortcut configuration into shortcut information.
+ * @param config Keyboard shortcut configuration
+ * @returns Parsed webview shortcuts
+ */
+export function parseWebviewKeyboardShortcutConfig(
+    config: WebviewKeyBindingConfiguration,
+): WebviewKeyBindings {
+    const webviewKeyBinding = {} as WebviewKeyBindings;
+    config = { ...getDefaultConfig(), ...config };
+    Object.keys(config).forEach((key) => {
+        const keyType = key as WebviewAction;
+        webviewKeyBinding[keyType] = getShortcutInfo(
+            config[key as keyof WebviewKeyBindingConfiguration],
+        );
+    });
+    return webviewKeyBinding;
+}
+
+/**
+ * Matches a raw keyboard event against the event generated from user configuration.
+ * @param event Raw keyboard event
+ * @param keyCombination parsed key combination from user configuration
+ * @returns True if the event matches the configuration, false otherwise
+ */
+export function eventMatchesShortcut(
+    event: KeyboardEvent | ReactKeyBoardEvent<any>,
+    combo: WebviewKeyCombination,
+): boolean {
+    if (!combo) return false;
+
+    // Treat undefined modifier as "don't care".
+    if (combo.ctrlKey !== undefined && combo.ctrlKey !== event.ctrlKey) return false;
+    if (combo.metaKey !== undefined && combo.metaKey !== event.metaKey) return false;
+    if (combo.altKey !== undefined && combo.altKey !== event.altKey) return false;
+    if (combo.shiftKey !== undefined && combo.shiftKey !== event.shiftKey) return false;
```

Comment:
For modifier keys, shouldn't we treat undefined modifiers as "must not be pressed"?